### DISCUSSION TITLE
Environment Variable Processors - Default String Value

### DISCUSSION THREAD

**Discussion by cancan101** [symfony/symfony#57781] [Q&A]
Are there any of the Environment Variable Processors that allow specifying a fallback string literal?
There is this:
```
env(default:fallback_param:BAR)
```
but `fallback_param` is a param rather than a literal string. Omitting `fallback_param` falls back to `null`.
I am looking for something like: `env(default_str:fallback_str:BAR)`

**Comment by 94noni** [comment]
Perhaps use https://symfony.com/blog/new-in-symfony-6-3-dx-improvements-part-2#build-parameters-in-service-container
?
What is your use case?

**Reply by cancan101** [reply]
Here is a good example of a use case: https://github.com/orgs/pimcore/discussions/17426#discussioncomment-10213337

I want to load a setting from the ENV but specific a static fallback.

**Comment by cancan101** [comment]
The simplest solution [per](https://symfony.com/doc/current/configuration.html#configuration-based-on-environment-variables) is to define the default value for an env var like this:
```yaml
# config/packages/framework.yaml
parameters:
    # if the SECRET env var value is not defined anywhere, Symfony uses this value
    env(SECRET): 'some_secret'

# ...
```
